src/Connection.php

```diff
@@ -1094,23 +1098,46 @@ public function getTransactionNestingLevel(): int
     }
 
     /**
-     * Returns the ID of the last inserted row, or the last value from a sequence object,
-     * depending on the underlying driver.
+     * Returns the ID of the last inserted row.
      *
-     * Note: This method may not return a meaningful or consistent result across different drivers,
-     * because the underlying database may not even support the notion of AUTO_INCREMENT/IDENTITY
-     * columns or sequences.
+     * If the underlying driver does not support identity columns, an exception is thrown.
      *
-     * @param string|null $name Name of the sequence object from which the ID should be returned.
+     * @return int|string The last insert ID, as an integer or a numeric string.
      *
-     * @return string A string representation of the last inserted ID.
+     * @throws Exception
+     */
+    public function lastInsertId()
+    {
+        try {
+            return $this->getWrappedConnection()->lastInsertId();
+        } catch (DriverException\IdentityColumnsNotSupported $e) {
+            throw new IdentityColumnsNotSupported($e->getMessage(), $e);
+        } catch (DriverException\NoIdentityValue $e) {
+            throw new NoIdentityValue($e->getMessage(), $e);
+        } catch (DriverException $e) {
+            throw $this->convertException($e);
+        }
+    }
+
+    /**
+     * Returns the current sequence value for the given sequence name.
+     *
+     * If the underlying driver does not support sequences, an exception is thrown.
+     *
+     * @param string $name The sequence name.
+     *
+     * @return int|string The sequence number, as an integer or a numeric string.
```

Comment:
I remember having discussions about what return type should these methods have but I don't remember if we came to any conclusion. It definitely cannot be an `int` because the value may not fit into `int32` but it could be a numeric `string` enforced at the method declaration level. Why `int|string`? Should we document the decision-making process?

Reply:
Please see https://github.com/doctrine/dbal/pull/3759#discussion_r353246669

Reply:
Thanks for digging it out. Let's have it declared as `int|string`.

Reply:
Alright, nothing to change here then! Resolving.